torch/_subclasses/functional_tensor.py

```diff
@@ -292,6 +292,10 @@ def __init__(self, pre_dispatch=False, export=False, _allow_token_discovery=Fals
         # track of the ordering between side effectful operations.
         self._tokens: Dict[Any, torch.Tensor] = {}
 
+        # Filled after forward tracing is done, see on_forward_tracing_end
+        self._tokens_forward_output: Dict[Any, torch.Tensor] = {}
```

Comment:
just a nit, but right now this PR "violates invariants" in both directions:
(1) the `create_joint()` logic needs to manually mess around with tokens from `FunctionalTensorMode`
(2) `FunctionalTensorMode` needs to be aware of a fw/bw context and treat tokens separately between the two

If we are already going to have to live with (1) (the joint-tracing code knowing about effect tokens), I would rather not also do (2). The easiest way to do that is probably just to have the `create_joint` function manually grab `mode._tokens` at the beginning vs end of the backward, and diff the two tokens lists to know which tokens were only part of the backward.